pkg/provision/providers/qemu/launch_darwin.go

```diff
@@ -6,9 +6,60 @@ package qemu
 
 import (
 	"context"
+	"fmt"
+	"net"
+	"net/netip"
 	"os/exec"
+	"strings"
+
+	"github.com/siderolabs/talos/pkg/provision"
+	"github.com/siderolabs/talos/pkg/provision/providers/vm"
 )
 
+type networkConfig struct {
+	networkConfigBase
+	StartAddr netip.Addr
+	EndAddr   netip.Addr
+}
+
+func getLaunchNetworkConfig(state *vm.State, clusterReq provision.ClusterRequest, nodeReq provision.NodeRequest) networkConfig {
+	// This ip will be assigned to the bridge
+	// The following ips will be assigned to the vms
+	startAddr := clusterReq.Nodes[0].IPs[0].Prev()
+	endAddr := clusterReq.Nodes[len(clusterReq.Nodes)-1].IPs[0].Next()
+
+	return networkConfig{
+		networkConfigBase: getLaunchNetworkConfigBase(state, clusterReq, nodeReq),
+		StartAddr:         startAddr,
+		EndAddr:           endAddr,
+	}
+}
+
+func getNetdevParams(networkConfig networkConfig, id string) string {
+	netDevArg := "vmnet-shared,id=" + id
+	cidr := networkConfig.CIDRs[0]
+	m := net.CIDRMask(cidr.Bits(), 32)
+	subnetMask := fmt.Sprintf("%d.%d.%d.%d", m[0], m[1], m[2], m[3])
+	netDevArg += fmt.Sprintf(",start-address=%s,end-address=%s,subnet-mask=%s", networkConfig.StartAddr, networkConfig.EndAddr, subnetMask)
+
+	return netDevArg
+}
+
+// getConfigServerAddr returns the ip accessible to the VM that will route to the config server.
+// hostAddrs is the address on which the server is accessible from the host network.
+func getConfigServerAddr(hostAddrs net.Addr, config LaunchConfig) (net.Addr, error) {
+	split := strings.Split(hostAddrs.String(), ":")
```

Comment:
this way doesn't work for IPv6 addresses which look like `[2000::1]:4000`, so please use https://pkg.go.dev/net#SplitHostPort

but in general, there's a better experience with https://pkg.go.dev/net/netip#AddrPort, so I would suggest to use it here, as we're dealing with IP and port, not a random string.

It looks like all we do is just replace the address